### ISSUE TITLE
Introducing coal plates into the compressor resets the compressor.

### ISSUE THREAD

**Issue by ivangouba** [Electrical-Age/ElectricalAge#490]
I have a setup where I have a macerator that is connected with EnderIO pipes to a compressor, which is connected with EnderIO pipes to another compressor to make diamonds. The problem is that each time a coal plate is introduced into the last compressor (the one that turns coal plates into diamonds), it starts the process again.
![2016-07-13_20 04 40](https://cloud.githubusercontent.com/assets/8239883/16812286/16a39d60-4935-11e6-9274-bdcccd8a98f8.png)

P.D. Using itemducts from Thermal Foundation doesn't produce the issue.

**Comment by cm0x4D** [comment]
Can you check if you have the same behavior if feeding the coal plates manually to the compressor, because this actually works for me...

**Comment by cm0x4D** [comment]
There are two ways to change an inventory slot content count:
- Increment the existing itemstack count
- Replace the existing itemstack with a new one with the same item type, but a different size

EnderIO seems to do the second, while adding items in the UI does the first. In the EA code we checked if the stack remains the same (object reference) and if not, the process is restarted. I changed the code to restart the process only if the actual type of item changes.

See #500. Still some testing and reviewing needed, but essentially the issue is fixed.
